bftengine/src/bcstatetransfer/BCStateTran.hpp

```diff
@@ -766,6 +767,12 @@ class BCStateTran : public IStateTransfer {
 
   FetchingState lastFetchingState_;
 
+  // For speed, we choose to make the API call isCollectingState() non-blocked, and it doesn't pass through handoff
+  // queue. This flag represents ST basic state to the outside: is it currently collecting/not collecting. It is set in
+  // slightly different time points to ensure thread safety towards external modules and threads.
+  std::atomic_bool externalCollectingState_{false};
```

Comment:
Why not simply name it isCollecting_?
The addition of "external" does not give additional information as there is no "internal" collection state.